src/settings.js

```diff
@@ -1022,9 +1022,13 @@ var BINARYEN_SCRIPTS = "";
 
 // Whether to ignore implicit traps when optimizing in binaryen.  Implicit traps
 // are the unlikely traps that happen in a load that is out of bounds, or
-// div/rem of 0, etc. We can reorder them, but we can't ignore that they have
-// side effects, so turning on this flag lets us do a little more to reduce code
-// size.
+// div/rem of 0, etc. With this option set, we assume that loads have no side
+// effects, for example, as the only side effect it can have is the possible
+// trap. This is *not* safe in general, as you may have a load behind a condition
+// and the condition ensures it is safe to do, but if we assumed the load never
```

Comment:
could be simpler as something like "behind a condition which ensures it it is safe; but if the load cannot have side effects it could be executed unconditionally."